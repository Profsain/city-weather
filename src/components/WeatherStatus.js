import React from 'react';
import { useSelector } from 'react-redux';
import { Grid } from '@mui/material';
import StatusCards from './StatusCards';
import '../styles/WeatherStatus.css';

const WeatherStatus = () => {
  const weatherData = useSelector((state) => state.weather.weatherData);
  const wind = weatherData?.wind?.speed || 0;
  const humidit = weatherData?.main?.humidity || 0;
  const visibilit = weatherData?.visibility || 0;
  const pressur = weatherData?.main?.pressure || 0;
  return (
    <div className="WeatherStatus">
      <h2>Today&apos;s Highlight</h2>
      <Grid
        container
        gap={6}
        direction="row"
        justifyContent="center"
        alignItems="center"
      >
        <Grid item xs={12} md={5}>
          <StatusCards
            title="Wind Status"
            measure={wind}
            unit="mph"
          />
        </Grid>
        <Grid item xs={12} md={5}>
          <StatusCards
            title="Humidity"
            measure={humidit}
            unit="%"
          />
        </Grid>
        <Grid item xs={12} md={5}>
          <StatusCards
            title="Visibility"
            measure={visibilit}
            unit="miles"
          />
        </Grid>
        <Grid item xs={12} md={5}>
          <StatusCards
            title="Air Pressure"
            measure={pressur}
            unit="mb"
          />
        </Grid>
      </Grid>
    </div>
  );
};

export default WeatherStatus;
