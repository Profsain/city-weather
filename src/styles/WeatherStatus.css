.WeatherStatus {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-bottom: 18px;
  margin-left: 38px;
}

.WeatherStatus h2 {
  padding: 6px 32px;
}
