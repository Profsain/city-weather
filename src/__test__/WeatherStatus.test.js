import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import WeatherStatus from '../components/WeatherStatus';

it('Renders WeatherStatus component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <WeatherStatus />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
