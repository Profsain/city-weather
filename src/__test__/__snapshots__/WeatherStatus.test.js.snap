// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders WeatherStatus component correctly 1`] = `
<div
  className="WeatherStatus"
>
  <h2>
    Today's Highlight
  </h2>
  <div
    className="MuiGrid-root MuiGrid-container css-188hzqz-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
    >
      <div
        className="StatusCard"
      >
        <h4
          className="Title"
        >
          Wind Status
        </h4>
        <p
          className="Record"
        >
          0
          <span>
            mph
          </span>
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
    >
      <div
        className="StatusCard"
      >
        <h4
          className="Title"
        >
          Humidity
        </h4>
        <p
          className="Record"
        >
          0
          <span>
            %
          </span>
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
    >
      <div
        className="StatusCard"
      >
        <h4
          className="Title"
        >
          Visibility
        </h4>
        <p
          className="Record"
        >
          0
          <span>
            miles
          </span>
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
    >
      <div
        className="StatusCard"
      >
        <h4
          className="Title"
        >
          Air Pressure
        </h4>
        <p
          className="Record"
        >
          0
          <span>
            mb
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
